### PR TITLE
Add ground test bridge and bridge driver

### PR DESCRIPTION
**Pull request by zhemao**
This bridge is needed to allow FireSimTraceGen to exit cleanly.

### CODE REVIEW

File: sim/firesim-lib/src/main/scala/bridges/GroundTestBridge.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+class GroundTestBridgeModule(implicit p: Parameters)
+    extends BridgeModule[HostPortIO[GroundTestBridgeTargetIO]] {
+  val io = IO(new WidgetIO)
+  val hPort = IO(HostPort(new GroundTestBridgeTargetIO))
+
+  val tFire = hPort.toHost.hValid && hPort.fromHost.hReady
+  hPort.toHost.hReady := tFire
+  hPort.fromHost.hValid := tFire
```
Comment (davidbiancolin) [inline sim/firesim-lib/src/main/scala/bridges/GroundTestBridge.scala]: Tie both of these off to true.B

File: sim/firesim-lib/src/main/scala/bridges/GroundTestBridge.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  genROReg(hPort.hBits.success, "success")
```
Comment (davidbiancolin) [inline sim/firesim-lib/src/main/scala/bridges/GroundTestBridge.scala]: Add an intermediate reg here that's sticky on success (i.e., set when toHost.fire && hPort.hBits.sucess) and tie toHost.hReady := true.B per previous comment.

File: sim/firesim-lib/src/main/scala/bridges/GroundTestBridge.scala (hunk 1 above)
Comment (davidbiancolin) [reply sim/firesim-lib/src/main/scala/bridges/GroundTestBridge.scala]: You don't produce output tokens. We need avoid this pattern of making the channels dependent on one another whenever possible.

File: sim/src/main/cc/firesim/firesim_top.cc [hunk 3]
```diff
@@ -331,6 +332,49 @@ uint64_t host_mem_offset = -0x80000000LL;
     #endif
 #endif
 
+#ifdef GROUNDTESTBRIDGEMODULE_struct_guard
```
Comment (davidbiancolin) [inline sim/src/main/cc/firesim/firesim_top.cc]: Technically you only need this you're going to super-node GroundTest but this is safe.

**Comment by zhemao** [comment]
Okay I updated the token handling and success register. @davidbiancolin Can you take another look?

**Pull request merged**